### ISSUE TITLE
Extension of generic type 'Array<T>' from a different module cannot provide public declarations

### ISSUE THREAD

**Issue by prbaron** [pNre/ExSwift#82]
Hello, here is my problem when using your library. This is my first Swift (and iOS) application so I could have made errors in importing or whatever.

CONTEXT
importing your library into my application by drag and dropping the ExSwift folder (not the xcodeproj) into the root of my iOS application.

ACTION
Build the application. No error in my personal code.

RESULT
errors in all methods of the Array class of ExSwift : "Extension of generic type 'Array<T>' from a different module cannot provide public declarations"

EXPECTED
The application should build.

Thanks in advance for your help. Great library by the way !

**Comment by pNre** [comment]
Hi,
this is a known problem with Xcode < 6.3 where Swift frameworks cannot declare public extensions.
As of the latest Xcode 6.3 beta this seems solved, so you have two options:
1. Use Xcode 6.3 to compile the project (but in that case you won't be able to publish anything on the App Store until 6.3 is released to the public)
2. Follow the installation steps for Xcode 6.1/6.2: https://github.com/pNre/ExSwift#installation

**Comment by prbaron** [comment]
I thought I followed the steps for Xcode 6.1.

Where do I should write this ? 
```
components.takeFirst() { $0.completed }
```

**Comment by pNre** [comment]
It's just an example, you don't have to write it anywhere.

**Comment by prbaron** [comment]
Ok, then I made the point 2. Should I make the point 1 two ? i do not see why it should be needed.

**Comment by prbaron** [comment]
Do I have to import ExSwift in the embedded libraries in the project manager ?

**Comment by pNre** [comment]
Nope, you have to add the source files from ExSwift to your project (the second point in the list from the readme).

**Comment by prbaron** [comment]
OK so do you know why do I have the error ?

**Comment by pNre** [comment]
You're probably still linking the framework.
Remove any reference to ExSwift from your project and just add the source files.

**Comment by prbaron** [comment]
Still the same project...

**Comment by feinstruktur** [comment]
"1. Use Xcode 6.3 to compile the project (but in that case you won't be able to publish anything on the App Store until 6.3 is released to the public)"

@pNre Does this work for you with 6.3b? I'm using Version 6.3 (6D554n) (beta4) and I can't manage to avoid this error. I must be holding it wrong - is there a way to make these declarations public so it can be built und used as a framework?

**Comment by pNre** [comment]
Can you please link an example project?

**Comment by vmanot** [comment]
What dark magic do you speak of @pNre? It is currently impossible to provide public extensions to generic classes from foreign modules. To till this, no workaround has been found. I hope you've got a "... yet" stored in there for me.

**Comment by feinstruktur** [comment]
Just add 'public' to an Array extension method like 'each' and you'll get the error, no sample project required :)

**Comment by pNre** [comment]
You're both right, this cannot be done (yet), I must have been drunk when I tried using it as module in beta 3.

**Comment by vmanot** [comment]
Swift 2.0 now allows extensions to generic types. Hooray!
